### DISCUSSION TITLE
Grain reconstruction for two phases in calcGrains()

### DISCUSSION THREAD

**Discussion by rrkamath** [mtex-toolbox/mtex#1353] [Ask Anything] [answered]
Hi all,

While performing grain reconstruction, I wanted to use two separate threshold angles to define boundaries for two different phases present in my sample. Is that possible? If yes, could you please let me know how to code it using the calcGrains function ? More details are shared below. Many thanks !

Best
Rakesh

My current grain reconstruction code section looks like this:

```
%% Grain reconstruction

% reconstruct the grain structure
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',2*degree);

% remove some very small features (10 pixels)
ebsd(grains(grains.grainSize<10)) = [];

% redo grain segementation
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',2*degree,'boundary','tight');
```

There are two phases present in the indexed pixels - Ti-hcp and Ti-bcc. I wanted to reconstruct Ti-hcp grains using a 2*degree threshold and the Ti-bcc phase using a 10*degree threshold.

**Comment by kilir** [comment]
Hi,
this is currently not possible. What is the limitation if two separate sets of grains are computed?
Cheers,
Rüdiger

**Reply by nyyssont** [reply]
One way to approach this would be to reconstruct all grains with a tight threshold and then merge the grains of a desired phase with a loose threshold. This is a workaround but might provide an equivalent result.

https://mtex-toolbox.github.io/GrainMerge.html

**Reply by rrkamath** [reply]
Thanks for the suggestion, Rüdiger. That makes perfect sense to me. I will give that a try.

One of the later goals of this exercise is to reconstruct the parent Ti-bcc from the microstructure which contains both child Ti-hcp and retained Ti-bcc. My thought was that having separate sets of grains will not include the retained Ti-bcc and therefore won't provide the fully reconstructed parent Ti-bcc microstructure. Does that make sense? On that note, can we add two separate sets of grains under one ebsd() function? In principle, that might take care of the worries I have on reconstruction (explained above), if at all that's a problem.

Best
Rakesh

**Reply by rrkamath** [reply]
> One way to approach this would be to reconstruct all grains with a tight threshold and then merge the grains of a desired phase with a loose threshold. This is a workaround but might provide an equivalent result.
> 
> https://mtex-toolbox.github.io/GrainMerge.html

I will look into the merge grains approach as well. Thank you for the suggestion, Tuomo!

Best
Rakesh

**Comment by ralfHielscher** [comment, marked as answer]
I have implemented this feature in 6c995ce5531160a6a96f87cb68a968e5166330c4 using the syntax
```matlab
grains = calcGrains(ebsd,'angle',{angle_1,angle_2,angle_3})
```
Ralf.

**Reply by rrkamath** [reply]
Thanks Ralf!
